src/network/protocol/grandpa_warp_sync.rs

```diff
@@ -32,7 +41,24 @@ pub enum DecodeGrandpaWarpSyncResponseError {
 // TODO: make this a zero-cost API
 pub fn decode_grandpa_warp_sync_response(
     bytes: &[u8],
-) -> Result<Vec<GrandpaWarpSyncResponseFragment>, DecodeGrandpaWarpSyncResponseError> {
+) -> Result<GrandpaWarpSyncResponse, DecodeGrandpaWarpSyncResponseError> {
+    nom::combinator::map(
+        nom::sequence::tuple((
+            decode_fragments,
+            // TODO: remove this `opt` when a polkadot version that serves
+            // `is_finished` is released.
+            nom::combinator::opt(nom::number::complete::le_u8),
+        )),
+        |(fragments, is_finished)| GrandpaWarpSyncResponse {
+            fragments,
+            is_finished: is_finished.map(|byte| byte == 1),
```

Comment:
```suggestion
            is_finished: is_finished.map(|byte| byte != 0),
```

Not that it matters too much